packages/client-config/src/client-config-contributor/custom_client_config_contributor.ts

```diff
@@ -0,0 +1,68 @@
+import { ClientConfigContributor } from './client_config_contributor.js';
+import { UnifiedBackendOutput } from '@aws-amplify/backend-output-schemas';
+import { CustomClientConfig } from '../client-config-types/client_config.js';
+
+// TODO: these types are copied, should be moved to platform-types perhaps.
+export type ClientConfigDestination = {
+  clientConfigFormat: string;
+  path: Array<string>;
+}
+
+export type CustomOutputOptions = {
+  clientConfigDestinations?: Array<ClientConfigDestination>
+}
+
+export class CustomClientConfigContributor implements ClientConfigContributor {
+  contribute = async ({
+    ['AWS::Amplify::Custom']: customOutputs,
+  }: UnifiedBackendOutput): Promise<
+    CustomClientConfig | Record<string, never>
+  > => {
+    if (
+      customOutputs === undefined ||
+      Object.keys(customOutputs.payload).length === 0
+    ) {
+      console.log('custom outputs undefined');
+      return {};
+    }
+
+    console.log(customOutputs);
+
+    const result: CustomClientConfig & Record<string, unknown> = { custom: {} };
+
+    Object.entries(customOutputs.payload).forEach(([key, value]) => {
+      const customOutputData = JSON.parse(value);
+      // TODO how do we do custom namespacing ??
```

Comment:
Is the concern here that a custom key name may collide with one of our internal names? CDK will fail if you try to add multiple CFN outputs with the same name. But if we want to allow this behavior, then yes we'd need some sort of namespacing. That could probably go as a prerequisite PR to the rest of this change.

Reply:
Yes, that is the concern.
Which also raises a question if we should be namespacing known verticals outputs. Today we have only auth and data but at some point there will be more.

Reply:
However, then if customer passes an output key to `backend.addCustomOutput` they'd expect it to see in CFN console I guess.

So perhaps we shouldn't namespace and let them figure out potential collisions and rely on CFN validation.
Regardless of that I think we should do namespacing for our verticals (at least those we own) to minimize collision probability.

Reply:
FWIW, the only output I have seen that has any chance of collision is "region". All of our other output names are highly domain-specific.

Personally I think it's fine to not namespace and if there's a collision synth will fail. Then you just change your output to a slightly different name and carry on.

Reply:
that solves custom outputs.

However, what I'm worried about is that some day 5th category is implemented by some team and never tested with `data`. Then attempt to use that new category with `data` will result in permanent conflict and p0 bug ...

Reply:
Yeah that's a fair callout